cmake/modules/FindHostTools.cmake

```diff
@@ -0,0 +1,125 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2022, Nordic Semiconductor ASA
+
+# FindHostTools module for locating a set of tools to use on the host for
+# Zephyr development.
+#
+# This module will lookup following tools for Zephyr development:
+# +---------------------------------------------------------------+
+# | Tool               | Required |  Notes:                       |
+# +---------------------------------------------------------------+
+# | Generic C-compiler | Yes      |  Pre-processing of devicetree |
+# | Zephyr-sdk         |          |                               |
+# | gperf              |          |                               |
+# | openocd            |          |                               |
+# | bossac             |          |                               |
+# | imgtool            |          |                               |
+# +---------------------------------------------------------------+
+#
+# The module defines the following variables:
+#
+# 'CMAKE_C_COMPILER'
+# Path to C compiler.
+# Set to 'CMAKE_C_COMPILER-NOTFOUND' if no C compiler was found.
+#
+# 'GPERF'
+# Path to gperf.
+# Set to 'GPERF-NOTFOUND' if gperf was not found.
+#
+# 'OPENOCD'
+# Path to openocd.
+# Set to 'OPENOCD-NOTFOUND' if openocd was not found.
+#
+# 'BOSSAC'
+# Path to bossac.
+# Set to 'BOSSAC-NOTFOUND' if bossac was not found.
+#
+# 'IMGTOOL'
+# Path to imgtool.
+# Set to 'IMGTOOL-NOTFOUND' if imgtool was not found.
+#
+# 'HostTools_FOUND', 'HOSTTOOLS_FOUND'
+#  True if all required host tools was found.
+
+include(extensions)
+
+if(HostTools_FOUND)
+  return()
+endif()
+
+# gperf is an optional dependency
+find_program(GPERF gperf)
+
+# openocd is an optional dependency
+find_program(OPENOCD openocd)
+
+# bossac is an optional dependency
+find_program(BOSSAC bossac)
+
+# imgtool is an optional dependency (the build may also fall back to scripts/imgtool.py
+# in the mcuboot repository if that's present in some cases)
+find_program(IMGTOOL imgtool)
+
+# Pick host system's toolchain if we are targeting posix
+if("${ARCH}" STREQUAL "posix")
+  if(NOT "${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "llvm")
+    set(ZEPHYR_TOOLCHAIN_VARIANT "host")
+  endif()
+endif()
+
+# Keep XCC_USE_CLANG behaviour for a while.
+if ("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "xcc"
+    AND "$ENV{XCC_USE_CLANG}" STREQUAL "1")
+  set(ZEPHYR_TOOLCHAIN_VARIANT xcc-clang)
+  message(STATUS "XCC_USE_CLANG is deprecated. Please set ZEPHYR_TOOLCHAIN_VARIANT to 'xcc-clang'")
+endif()
+
+if(NOT ZEPHYR_TOOLCHAIN_VARIANT AND
+   (CROSS_COMPILE OR (DEFINED ENV{CROSS_COMPILE})))
+    set(ZEPHYR_TOOLCHAIN_VARIANT cross-compile)
+endif()
+
+find_package(Zephyr-sdk 0.15)
```

Comment:
In CamelCase, `ZephyrSdk` would be a more appropriate name.

Reply:
yes, but I would like to have this file aligned with the Zephyr-sdk package from here:
https://github.com/zephyrproject-rtos/sdk-ng/blob/main/cmake/Zephyr-sdkConfig.cmake

so if we adjust `Zephyr-sdk` to `ZephyrSdk` then I will happily change.